src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -144,41 +144,48 @@ final class TrialService
       case None => None
       case Some(status) => status.state
     }
+
     if (innerState.isEmpty || innerState != newState.state) {
-      // TODO throw bad request or invalid state transition exception
-      require(newState.state.nonEmpty, "Cannot transition to an unspecified state")
-      assert(newState.state.get.isAllowedFrom(innerState), s"Cannot transition from $currentState.get.state to $newState.state")
+      if (newState.state.isEmpty || !newState.state.get.isAllowedFrom(innerState))
+        throw new FireCloudException(s"Cannot transition from ${innerState} to $newState.state.get")
       true
     } else {
       false
     }
   }
 
-  private def getUserSubjectId(userEmail: String): Future[String] = {
-    // TODO: Handle unregistered users which get 404 from Sam causing adminGetUserByEmail to throw
-    // TODO: Handle errors that may come up while querying Sam
-    samDao.adminGetUserByEmail(RawlsUserEmail(userEmail)).map(_.userInfo.userSubjectId)
-  }
-
   private def executeStateTransitions(managerInfo: UserInfo, userEmails: Seq[String],
                                       statusTransition: (WorkbenchUserInfo, Option[UserTrialStatus]) => UserTrialStatus,
                                       transitionPostProcessing: (Attempt, Option[UserTrialStatus], UserTrialStatus) => Unit): Future[PerRequestMessage] = {
     var results: Seq[(String, String)] = Seq()
     userEmails.foreach { userEmail =>
-      val finalStatus = getUserSubjectId(userEmail) flatMap { subId =>
+      val finalStatus = samDao.adminGetUserByEmail(RawlsUserEmail(userEmail)) flatMap { regInfo =>
```

Comment:
Do we have negative tests for this Sam call? We may get an exception or `404`. Not sure if there are others.